app/code/Magento/Customer/Model/ResourceModel/Grid/Collection.php

```diff
@@ -176,4 +179,18 @@ private function getRegionNameExpression(): \Zend_Db_Expr
             $defaultNameExpr
         );
     }
+
+    /**
+     * Returns condition array data
+     *
+     * @param string $field
+     * @param string $value
+     * @return string[]
+     */
+    private function getCondition(string $field, string $value): array
+    {
+        return $field === 'email' && preg_match(self::EXACT_SEARCH_PATTERN, $value)
```

Comment:
I believe it would be better to trim double quotes from all values not only for email. Also, we can remove heavy preg_match operation and trim values without conditions.